Service configuration and best practices (#1055)
vtexdocs/dev-portal-content: docs/vtex-io/Reference/concepts/vtex-io-documentation-service.md

Let's delve into the parameters you can set within the `service.json` file:

|Name|Type|Description|
|--|--|--|

> |`memory`|`number`|The specified memory size (in MB) allocated to the service. Default: 128, maximum: 1024.|

Reviewer: ```suggestion
|`memory`|`number`|Defines the memory size (in MB) that should be allocated to the service. Default: 128, maximum: 1024.|
```

Reviewer: Para ficar padronizado em relação às outras descrições

|`↳public`|`boolean`|Defines if the route is public or private. If `true`, the route will be available at `{account}.myvtex.com` and `{account}.vtexcommercestable.com.br`. If `false`, the route will be at `app.io.vtex.com/{vendor}.{appName}/v{majorVersion}/{account}/{workspace}` and will require a `VtexidClientAutCookie` for [authentication](https://developers.vtex.com/docs/guides/api-authentication-using-user-tokens). Using `appKey` and `appToken` for authentication will not work.|
|`↳access`|`string`|Possible values: `public`, `authenticated`, or `authorized`.|
|`↳policies`|`array of objects`|Defines the actions allowed or denied with the route by whom. More details on [Resource-based Policies](https://developers.vtex.com/docs/guides/vtex-io-documentation-policies#resource-based-policies).|
|`↳rateLimitPerReplica`|`object`|This object contains parameters whose values define specific throttling limits for the route. The values of this object override the global `rateLimitPerReplica` values.|
|`↳↳perMinute`|`number`|Defines the maximum number of requests allowed per minute per replica for this route. This value overrides the global `rateLimitPerReplica.perMinute` value.|

> |`↳↳concurrent`|`number`|Defines the maximum number of requests a replica will handle for this route. This value overrides the global `rateLimitPerReplica.concurrent` value.|

Reviewer: Coloca em ordem alfabética para ficar mais fácil encontrar os parâmetros? 🙏🏼

In scenarios where an application has a low timeout setting and takes too long to respond, the platform will respond to the request with a timeout response and the data from the app will not be delivered to the client when the request could be processed. On the other hand, in scenarios where the timeout setting is high and the application usually does not take long to respond, if a problem occurs with the processing of the request, the platform will take too long to respond with a timeout and inform the client that a problem has occurred.

Balancing between a low or high timeout setting depends on the time that is expected to deliver a response and how fast the requester requires the response to perform its work.

Testing response times to decide a proper timeout setting can be done in various ways including, but not limited to, different connection conditions (mobile vs. tethered, instabilities, etc.), requests from different locations, different workloads and settings (memory, replicas, etc.), and measuring percentiles of the slowest response times.

### Replica management

> VTEX has an autoscaling feature for the IO infrastructure, which dynamically controls the number of instances of each app obeying certain parameters. We call replicas the number of instances of an app running in our infrastructure. The minimum and maximum replicas an app can have are determined by the parameters `minReplicas` and `maxReplicas` respectively. The current replicas increase and decrease automatically according to how much the app is requested in the platform, considering all the accounts where the app is installed.

Reviewer: ```suggestion
VTEX has an autoscaling feature for the IO infrastructure, which dynamically controls the number of instances of each app obeying certain parameters. We call replicas the number of instances of an app running in our infrastructure. The minimum and maximum replicas an app can have are determined by the parameters `minReplicas` and `maxReplicas`, respectively. The current replicas increase and decrease automatically according to how much the app is requested in the platform, considering all the accounts where the app is installed.
```

> The values of `minReplicas` and `maxReplicas` should be defined considering the expected demand for the app. If you have observability on the accounts where the app is installed, some metrics can help define the parameters, e.g. the number of accesses or purchases and how they fluctuate.

Reviewer: ```suggestion
The values of `minReplicas` and `maxReplicas` should be defined considering the expected demand for the app. If you have observability on the accounts where the app is installed, some metrics can help define the parameters, e.g., the number of accesses or purchases and how they fluctuate.
```
